src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CreateComponentCodeActionResolver.cs

```diff
@@ -32,14 +32,18 @@ internal class CreateComponentCodeActionResolver : RazorCodeActionResolver
 
         public override string Action => LanguageServerConstants.CodeActions.CreateComponentFromTag;
 
-        public override async Task<WorkspaceEdit> ResolveAsync(JObject data, CancellationToken cancellationToken)
+        public override async Task<WorkspaceEdit> ResolveAsync(object data, CancellationToken cancellationToken)
         {
             if (data is null)
             {
                 return null;
             }
 
-            var actionParams = data.ToObject<CreateComponentCodeActionParams>();
+            if (!(data is CreateComponentCodeActionParams actionParams))
+            {
+                actionParams = (data as JObject)?.ToObject<CreateComponentCodeActionParams>();
+            }
+
             var path = actionParams.Uri.GetAbsoluteOrUNCPath();
```

Comment:
`actionParams` is still not guaranteed to not be null here correct?

Reply:
Yep. Added additional null checks. Thanks.